tests/ui/typeck/issue-100285.stderr

```diff
@@ -12,47 +12,11 @@ LL | |
 LL | |     }
    | |_____^ expected `i32`, found `()`
    |
-note: the function expects a value to always be returned, but loops might run zero times
-  --> $DIR/issue-100285.rs:2:5
-   |
-LL |     for i in 0..0 {
-   |     ^^^^^^^^^^^^^ this might have zero elements to iterate on
```

Comment:
Same here. We should even be detecting explicitly known to never iterate ranges, where the start and the end of a `..` range are the same, because the entire thing is *for sure* dead code.

Reply:
Yes, it would be a good idea to warn the user that `for i in 0..0 { ... }` is dead code. Maybe I can do that in a different PR.